commerce_attribute.module

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ * @file
+ * Hooks and functionality for the Commerce Attribute module.
+ */
+
+use Drupal\entity\BundleFieldDefinition;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_entity_type_alter().
+ */
+function commerce_attribute_entity_type_alter(&$entity_types) {
+  $entity_types['commerce_product_attribute']->setClass('Drupal\commerce_attribute\Entity\ProductAttribute');
+  $entity_types['commerce_product_attribute_value']->setClass('Drupal\commerce_attribute\Entity\ProductAttributeValue');
+  $entity_types['commerce_product_attribute']->setFormClass('add', 'Drupal\commerce_attribute\Form\ProductAttributeForm');
+  $entity_types['commerce_product_attribute']->setFormClass('edit', 'Drupal\commerce_attribute\Form\ProductAttributeForm');
+  $entity_types['commerce_product_attribute_value']->setFormClass('add', 'Drupal\commerce_attribute\Form\ProductAttributeValueForm');
+  $entity_types['commerce_product_attribute_value']->setFormClass('edit', 'Drupal\commerce_attribute\Form\ProductAttributeValueForm');
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function commerce_attribute_commerce_product_attribute_value_update($attribute_value) {
+
+  // Proceed only if price trait is enabled.
+  if (!$attribute_value->hasField('price')) {
+    return;
+  }
+  // Check if the attribute price is changed or not.
+  // If its not changed, we do not have to perform the variation
+  // price update process.
+  $attribute_value_original = $attribute_value->original;
+  $original_price = $attribute_value_original->get('price')->first()->toPrice()->getNumber();
+  $new_price = $attribute_value->get('price')->first()->toPrice()->getNumber();
+  if ($new_price == $original_price) {
+    return;
+  }
+
+  // Updates the variation price associated with the attribute value.
+  $price_handler = \Drupal::service('commerce_attribute.variation_price_handler');
+  $price_handler->updateProductVariationsPrice($attribute_value);
```

Comment:
**CHANGE REQUEST**

You don't need to declare a variable you're not going to use. With the suggested method name change, that would just be:

`\Drupal::service('commerce_attribute.variation_price_handler')->updatePrices($attribute_value);`

Reply:
Done